druntime/src/__builtins_msvc.d

```diff
@@ -0,0 +1,13488 @@
+   Authors: Harry Gillanders
+   Source: $(DRUNTIMESRC __builtins_msvc.d) */
+
+module __builtins_msvc;
+
+version (CRuntime_Microsoft)
+{
+    version = MSVCIntrinsics;
+}
+
+version (MSVCIntrinsics)
+{
+    version (X86)
+    {
+        version = X86_64_Or_X86;
+    }
+    else version (X86_64)
+    {
+        version = X86_64_Or_X86;
+        version = X86_64_Or_AArch64;
+        version = X86_64_Or_AArch64_Or_ARM;
+    }
+    else version (AArch64)
+    {
+        version = X86_64_Or_AArch64;
+        version = X86_64_Or_AArch64_Or_ARM;
+        version = AArch64_Or_ARM;
+    }
+    else version (ARM)
+    {
+        version = X86_64_Or_AArch64_Or_ARM;
+        version = AArch64_Or_ARM;
+    }
+
+    version (D_InlineAsm_X86)
+    {
+        version = InlineAsm_X86_64_Or_X86;
+    }
+    else version (D_InlineAsm_X86_64)
+    {
+        version = InlineAsm_X86_64_Or_X86;
+    }
+
+    version (LDC)
+    {
+        version = LDC_Or_GNU;
+
+             version (X86_64_Or_X86) private enum gccBuiltins = "ldc.gccbuiltins_x86";
+        else version (ARM) private enum gccBuiltins = "ldc.gccbuiltins_arm";
+        else version (AArch64) private enum gccBuiltins = "ldc.gccbuiltins_aarch64";
+    }
+    else version (GNU)
+    {
+        version = LDC_Or_GNU;
+
+        private enum gccBuiltins = "gcc.builtins";
+    }
+
+    import core.atomic : MemoryOrder;
+
+    static if (__traits(compiles, () {import core.simd : float4;}))
+    {
+        import core.simd : byte16, float4, long2, int4, ubyte16;
+
+        version (X86_64_Or_X86)
+        {
+            import core.simd : double2;
+        }
+
+        private enum canPassVectors = true;
+    }
+    else
+    {
+        private enum canPassVectors = false;
+    }
+
+    version (LDC)
+    {
+        version (X86_64_Or_X86)
+        {
+            pragma(LDC_intrinsic, "llvm.x86.sse2.pause")
+            private void __builtin_ia32_pause() @safe pure nothrow @nogc;
+
+            pragma(LDC_intrinsic, "llvm.x86.rdpmc")
+            private long __builtin_ia32_rdpmc(int) @safe nothrow @nogc;
+
+            pragma(LDC_intrinsic, "llvm.x86.rdtsc")
+            private long __builtin_ia32_rdtsc() @safe nothrow @nogc;
+
+            pragma(LDC_intrinsic, "llvm.x86.wbinvd")
+            private void __builtin_ia32_wbinvd() @safe nothrow @nogc;
+        }
+        else version (AArch64)
+        {
+            pragma(LDC_intrinsic, "llvm.aarch64.dmb")
+            private void __builtin_arm_dmb(int) @safe pure nothrow @nogc;
+
+            pragma(LDC_intrinsic, "llvm.aarch64.hint")
+            private void llvm_arm_hint(int) @safe pure nothrow @nogc;
+        }
+        else version (ARM)
+        {
+            pragma(LDC_intrinsic, "llvm.arm.dmb")
+            private void __builtin_arm_dmb(int) @safe pure nothrow @nogc;
+
+            pragma(LDC_intrinsic, "llvm.arm.hint")
+            private void llvm_arm_hint(int) @safe pure nothrow @nogc;
+        }
+    }
+    else version (GNU)
```

Comment:
Thanks for taking the time to consider both ldc and gdc here, it's really appreciated

 I was just wondering - GCC doesn't support MSVC. So are these GNU version blocks dead on Mingw targets?

Reply:
> Thanks for taking the time to consider both ldc and gdc here, it's really appreciated

:)
> So are these GNU version blocks dead on Mingw targets?

I think so. Unless GDC defines `CRuntime_Microsoft` when targeting MinGW?
I don't really know the story as far as GDC's support of Windows goes. So, I just implemented the GDC versions to save some other poor sod the hassle of doing it later it on.

I don't think there's anything preventing this implementation from working when targeting MinGW or Cygwin, if that's wanted?